sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/VirtualMachineOperationsTests.java

```diff
@@ -2053,6 +2055,123 @@ public void canUpdateVMWithoutEncryptionAtHost() {
         Assertions.assertFalse(vm.isEncryptionAtHost());
     }
 
+    @Test
+    public void canEnableWriteAccelerator() {
+        // required to be run on "Azure SDK Test Resources" subscription
+
+        // ref https://learn.microsoft.com/azure/virtual-machines/how-to-enable-write-accelerator
+        // 8 CPU likely to be the smallest VM that supports write accelerator on disks.
+        // only 1 disk is allowed to have write accelerator for M8
+        final String vmSize = "Standard_M8-2ms";
+        final int diskSize = 127;
+
+        Network network = this.networkManager.networks()
+            .define(vmName + "Network")
+            .withRegion(region)
+            .withNewResourceGroup(rgName)
+            .withAddressSpace("10.0.0.0/28")
+            .defineSubnet("default")
+            .withAddressPrefix("10.0.0.0/29")
+            .attach()
+            .create();
+
+        VirtualMachine vm = computeManager.virtualMachines()
+            .define(vmName)
+            .withRegion(region)
+            .withNewResourceGroup(rgName)
+            .withExistingPrimaryNetwork(network)
+            .withSubnet("default")
+            .withPrimaryPrivateIPAddressDynamic()
+            .withoutPrimaryPublicIPAddress()
+            .withPopularLinuxImage(KnownLinuxVirtualMachineImage.UBUNTU_SERVER_20_04_LTS)
+            .withRootUsername("tirekicker")
+            .withSsh(sshPublicKey())
+            // data disk
+            .withNewDataDisk(diskSize, 0,
+                new VirtualMachineDiskOptions().withDeleteOptions(DeleteOptions.DELETE)
+                    .withStorageAccountTypes(StorageAccountTypes.PREMIUM_LRS)
+                    .withCachingTypes(CachingTypes.READ_ONLY)
+                    .withWriteAcceleratorEnabled(true))
+            // os disk
+            .withOSDiskStorageAccountType(StorageAccountTypes.PREMIUM_LRS)
+            .withOSDiskDeleteOptions(DeleteOptions.DELETE)
+            .withOSDiskCaching(CachingTypes.READ_ONLY)
+            .withOSDiskWriteAcceleratorEnabled(false)
+            .withSize(vmSize)
+            .create();
+
+        Assertions.assertFalse(vm.isOsDiskWriteAcceleratorEnabled());
+        Assertions.assertTrue(vm.dataDisks().values().iterator().next().isWriteAcceleratorEnabled());
+
+        Disk dataDisk = computeManager.disks()
+            .define(vmName + "Disk1")
+            .withRegion(region)
+            .withExistingResourceGroup(rgName)
+            .withData()
+            .withSizeInGB(diskSize)
+            .withSku(DiskSkuTypes.PREMIUM_LRS)
+            .create();
+
+        vm.update().withoutDataDisk(0).apply();
+        vm.update()
+            .withDataDiskDefaultDeleteOptions(DeleteOptions.DETACH)
+            .withDataDiskDefaultCachingType(CachingTypes.NONE)
+            .withDataDiskDefaultWriteAcceleratorEnabled(true)
+            .withExistingDataDisk(dataDisk, 1, CachingTypes.NONE)
```

Comment:
What happens if we leave writeAccelerator empty and attach the data disk to the previous lun0? Will the newly attached data disk be write accelerator enabled?

Reply:
Yeah, I think it works this way too.

The reason I am using lun=1 was that if I remove and add in same PATCH, backend would think I was changing the createOption of the disk and fails.
But eventually I cannot do this in same PATCH anyway, the reason is that backend would think I have "2" write accelerator disk (even that I removed the lun 0).

M8 only allow 1 write accelerator disk in VM.